File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Module/Notification/NotificationRequests.swift
import Foundation

/// The notification request surface attached to one Apple runtime host.
@MainActor
public protocol NotificationRequests: AnyObject {
  /// Post one notification through the Apple host.
  func postNotification(
    _ request: RuntimeHostNotificationRequest
  ) -> UInt32

  /// Cancel one posted notification through the Apple host.
  func cancelNotification(
    identifier: String
  ) -> UInt32

  /// Cancel every posted notification through the Apple host.
  func cancelAllNotifications() -> UInt32
}

File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/module/notification/NotificationRequests.kt
package dev.destack.runtime.android.module.notification

/**
 * The notification request surface attached to one Android runtime host.
 */
public interface NotificationRequests {
    /**
     * Post one notification through the Android host.
     */
    public fun postNotification(
        request: RuntimeHostNotificationRequest,
    ): Int

    /**
     * Cancel one posted notification through the Android host.
     */
    public fun cancelNotification(
        identifier: String,
    ): Int

    /**
     * Cancel every posted notification through the Android host.
     */
    public fun cancelAllNotifications(): Int
}
